src/ubx/bullet.c

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: LGPL-3.0
+#include <sim2b/types/bullet_ubx.h>
+
+#include <sim2b/types/bullet.h>
+#include <sim2b/functions/bullet.h>
+
+struct sim2b_bullet_ubx_private {
+  struct sim2b_bullet_ubx_port_cache ports;
+  struct sim2b_bullet_nbx nblock;
+};
+
+int sim2b_bullet_ubx_init(ubx_block_t *b) {
+  int ret = -1;
+  struct sim2b_bullet_ubx_private *private;
+  if ((private = calloc(1, sizeof(struct sim2b_bullet_ubx_private))) == NULL) {
+    ubx_err(b, "sim2b_bullet: failed to alloc memory");
+    ret = EOUTOFMEM;
+    goto out;
+  }
+  b->private_data = private;
+
+  struct sim2b_bullet_nbx *nblock = &private->nblock;
+
+  // Get configuration information.
+  const int *nr_joints_temp;
+  int nr_joints_len = cfg_getptr_int(b, "nr_joints", &nr_joints_temp);
+  if (nr_joints_len != 1) {
+    ubx_err(b, "sim2b_bullet: nr_joints must be supplied");
+    goto out;
+  }
+  nblock->nr_joints = *nr_joints_temp;
+
+  const double *gravity_temp;
+  int gravity_len = cfg_getptr_double(b, "gravity", &gravity_temp);
+  if (gravity_len == 3) {
+    memcpy(nblock->grav, gravity_temp, gravity_len*sizeof(double));
```

Comment:
Would you mind adding spaces around the multiplication operator (also in the lines below)?
`gravity_len * sizeof(double)`

Reply:
Done (not yet pushed).